File: src/app/Models/CampaignFund.php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Model;
use Illuminate\Database\Eloquent\Relations\BelongsTo;

class CampaignFund extends Model
{
    protected $fillable = [
        'campaign_id',
        'amount_in',
        'amount_out',
        'description',
        'month',
        'year',
        'balance',
    ];

    public function campaign(): BelongsTo
    {
        return $this->belongsTo(Campaign::class);
    }
}

File: backend/internal/model/campaign_fund.go
package model

import (
	"time"

	"github.com/google/uuid"
)

type CampaignFund struct {
	ID          uuid.UUID `gorm:"type:uuid;primaryKey;default:gen_random_uuid()" json:"id"`
	CampaignID  uuid.UUID `gorm:"type:uuid;not null;index" json:"campaign_id"`
	AmountIn    int64     `gorm:"default:0" json:"amount_in"`
	AmountOut   int64     `gorm:"default:0" json:"amount_out"`
	Description string    `gorm:"type:text" json:"description"`
	Month       int       `json:"month"`
	Year        int       `json:"year"`
	Balance     int64     `gorm:"default:0" json:"balance"`
	CreatedAt   time.Time `json:"created_at"`
	UpdatedAt   time.Time `json:"updated_at"`

	Campaign Campaign `gorm:"foreignKey:CampaignID;constraint:OnDelete:CASCADE" json:"campaign,omitempty"`
}
